### DISCUSSION TITLE
Whiteflag encode/ decode

### DISCUSSION THREAD

**Discussion by MOsterhaus1** [fennelLabs/fennel-service-api#27] [Q&A]
In the body of the endpoint `/v1/whiteflag/encode/` is a property "messageCode". Does this property match with the codes of the functional messages as described in section 2.4.2 of the [Whiteflag protocol](https://standard.whiteflagprotocol.org/v1/)? Or with something else? I get an `502 bad error` as response when I use another letter than "A".

**Comment by MOsterhaus1** [comment]
Anther question regarding decoding. Is it right that the endpoint `/v1/whiteflag/decode/` doesn't return all the properies that are given in the `/v1/whiteflag/encode/` endpoint?

**Reply by Romulus10** [reply]
This is expected behavior. The contents of a Whiteflag message change slightly depending on which kind of message you're sending, so the decoder skips fields that aren't included in the specific hex packet you're decoding.

**Reply by MOsterhaus1** [reply]
Are there some fields that arn't encoded than also? Because when I send 

```
{
    "encryptionIndicator": "1",
    "duressIndicator": "0",\
    "messageCode": "A",   
    "referenceIndicator": "0",       
    "referencedMessage": "0000000000000000000000000000000000000000000000000000000000000000",
    "verificationMethod": "1",
    "verificationData": "https://organisation.int/whiteflag",
    "cryptoDataType": "0",
    "cryptoData": "",
    "text": "HQ",
    "resourceMethod": "",
    "resourceData": "",
    "pseudoMessageCode": "",
    "subjectCode": "52",
    "datetime": "2023-07-19T15:48:67Z",
    "duration": "",
    "objectType": "",
    "objectLatitude": "52.10002",
    "objectLongitude": "5.06496",
    "objectSizeDim1": "",
    "objectSizeDim2": "",
    "objectOrientation": "",
    "objectTypeQuant": ""
}
```

to the encode endpoint, the decodes message only contains 

```
{
  "prefix": "WF",
  "version": "1",
  "encryptionIndicator": "1",
  "duressIndicator": "0",
  "messageCode": "A",
  "referenceIndicator": "0",
  "referencedMessage": "0000000000000000000000000000000000000000000000000000000000000000",
  "verificationMethod": "1",
  "verificationData": "https://organisation.int/whiteflag"\
}
```

So I'm missing some fields that are filled in the encode endpoint, but that are not returned in the decode endpoint (text, subjectCode, dateTime, objectLatidtude, objectLongitude). Are those not encoded then, or are they not send back after decoding?

**Reply by Romulus10** [reply]
It looks like the problem here is that it sees that the `messageCode` is `A`, so it ignores fields that aren't part of the authentication message type. I'm checking in to the problem you mentioned above RE: changing the messageCode, which will help clear this up.

**Comment by Romulus10** [comment]
@MOsterhaus1 I just shipped a patch to both the Whiteflag backend and to the Fennel API that addresses these problems. See https://github.com/fennelLabs/fennel-service-api/discussions/31 for more details.
